build_tools/third_party/ruy/CMakeLists.txt

```diff
@@ -12,25 +12,72 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# A handwritten port of ruy's Bazel BUILD and *.bzl files.
+
 set(RUY_SRC_ROOT
   "${IREE_ROOT_DIR}/third_party/ruy/ruy/"
 )
 
-# TODO(benvanik): if android_arm then add -O3/-mfpu=neon.
-# TODO(benvanik): if andoird_arm64 then add -O3
 iree_select_compiler_opts(RUY_COPTS_BASE
   ALL
     "-I${IREE_ROOT_DIR}/third_party/ruy"
-  CLANG_OR_GCC
-    "-O3"
-    "-Wno-shadow"
-    "-Wno-missing-noreturn"
-    "-Wno-unused-local-typedef"
 )
-set(RUY_COPTS_AVX2)
-set(RUY_COPTS_AVX_VNNI)
-set(RUY_COPTS_SSE42)
-set(RUY_COPTS_SKYLAKE)
+
+# Some clang-based Windows toolchains have more aggressive interpretations
+# of -Wextra.
+if(NOT CMAKE_SYSTEM_NAME STREQUAL Windows)
+  iree_select_compiler_opts(RUY_COPTS_WARNINGS
+    CLANG_OR_GCC
+      -Wall
+      -Wextra
+      -Wc++14-compat
+      -Wundef
+  )
+endif()
+
+if(CMAKE_SYSTEM_PROCESSOR STREQUAL arm)
+  # This is assuming modern toolchains where no -mfloat-abi flag is
+  # needed to actually enable NEON.
+  set(RUY_COPTS_NEON -mfpu=neon)
+endif()
+
+set(RUY_COPTS
+  ${RUY_COPTS_BASE}
+  ${RUY_COPTS_WARNINGS}
+  ${RUY_COPTS_NEON}
+)
+
+if(CMAKE_SYSTEM_PROCESSOR STREQUAL x86_64 OR
```

Comment:
apparently it's AMD64 (sometimes); this seems to be the recommended way to handle it:
```
string(TOLOWER "${CMAKE_SYSTEM_PROCESSOR}" _CMAKE_SYSTEM_PROCESSOR_LOWER)
if(_CMAKE_SYSTEM_PROCESSOR_LOWER STREQUAL x86_64 OR
   _CMAKE_SYSTEM_PROCESSOR_LOWER STREQUAL amd64)
```